### PR TITLE
add submenu support to resource action menus

### PR DESCRIPTION
**Pull request by christianvogt**
Fixes https://issues.redhat.com/browse/ODC-2712
Related to work in topology view to have a way to access the Add actions in-context: https://github.com/openshift/console/pull/3918

### Problem
Kebab actions only support a flat list.
Patternfly does not have any sub menu support either.
Many more actions are being added to resources and a flat list of actions will become much too large.

### Solution
This PR builds off the earlier support for popper.js based menus. These menus automatically align to the left or right depending on where space is available. The preference is set to always open to the right if there is available space. This is why in the screenshots for the kebab and actions dropdown, the 2nd nested menu appears out to the right. The nesting of sub menus multiple levels deep is shown in the screenshots because the solution has no restriction on depth.

There is some duplicate code between kebab and topology in the popper setup but they are are based on two very different menu implementations. Also, we are trying to keep topology pure from console dependencies to eventually push it to PF.

The way to specify actions to appear in a sub menu is by using the new `path` prop of the `KebabOption`. The path is a `/` separated string where each segment denotes a new sub menu entry. Eg. `Menu 1/Menu 2/Menu 3`

There is keyboard support as well.
- `Right-Arrow | Enter` opens a sub menu
- `Left-Arrow` closes a sub menu
- `Esc` continues to close the entire menu for kebab and topology while only closing sub menus in the dropdown. This is because of the way dropdowns have a very different implementation.

image

image

image

Keyboard navigation:
submenus

cc @spadgett @karthikjeeyar @andrewballantyne

### CODE REVIEW

File: frontend/packages/topology/src/components/contextmenu/ContextSubMenuItem.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+              if (!subMenuRef.current || !subMenuRef.current.contains(e.relatedTarget as Node)) {
+                setOpen(false);
+              }
+            }}
+            onKeyDown={(e) => {
+              // open the sub menu on enter or right arrow
+              if (e.keyCode === 39 || e.keyCode === 13) {
+                setOpen(true);
+                e.stopPropagation();
+              }
```
Comment (abhinandan13jan) [inline frontend/packages/topology/src/components/contextmenu/ContextSubMenuItem.tsx]: should there be a keycode 37 check for closing the submenu?

Comment (christianvogt) [reply frontend/packages/topology/src/components/contextmenu/ContextSubMenuItem.tsx]: Left arrow is being captured in the scope of the sub menu itself. See code below on line 80 of this file.

**Comment by karthikjeeyar** [comment]
Verified locally
/lgtm

**Pull request merged**